eis_toolkit/vector_processing/proximity_tool.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import geopandas as gpd
+from numbers import Number
+from beartype import beartype
+from beartype.typing import Union
+from rasterio import profiles, transform
+from eis_toolkit.vector_processing.distance_computation import distance_computation
+
+@beartype
+def calculate_proximity(geodataframe: gpd.GeoDataFrame, raster_profile: Union[profiles.Profile, dict], maximum_distance: Number) -> np.ndarray:
+
+    """ Interpolates the distance values calculated by the distance_computation function between 0 and 1.
+        1 denots the value inside the polygon and 0 at the maximum distance.
+        If maximum_distance value is not provided, the program sets this value to the maximum value 
+        in the provided distance matrix.
+        Uses linear interpolation to calculate the distance from the polygon.
+
+        Args:
+            geodataframe: The GeoDataFrame with geometries to determine distance to.
+            raster_profile: The raster profile of the raster in which the distances
+                            to the nearest geometry are determined.
+            max_distance: The maximum distance in the output array.
+
+        Returns:
+            A 2D numpy array with the the distance values inverted.
+    """
+
+    out_matrix = distance_computation(geodataframe, raster_profile, maximum_distance)
+
+    minimum = np.min(out_matrix)
+    difference = maximum_distance - minimum
+    out_matrix = maximum_distance - out_matrix
+    out_matrix = out_matrix/difference
```

Comment:
I believe you could utilize `_min_max_scaling` from EIS Toolkit although I am not 100% sure. Please check and if it fits, we could replace this piece of code with a call to that function to have the operation centralized in one place only. 

There's also the point that if we extend the scaling options in the future, it is good to use our own `transformations` module wherever we can

Reply:
I tried using the _min_max_scaling, using this tool I could not achieve the inversion we intend to achieve. Moreover, I cannot see the scaling happening as well. Below I have attached the pictures.

![image](https://github.com/user-attachments/assets/8277ae69-93e8-4e47-947b-fa31c5a82cfd)

Reply:
What parameters did you use for min-max scaling? I noticed that that the tool does not allow `new_range=[(1, 0)]` right now, but you can try commenting out the lines 167-169 in `min_max_scaling.py` and then run it with `new_range=[(1, 0)]`. I think it should produce the correct results, just make sure you apply the max distance cut before the scaling operation.

Reply:
> I believe you could utilize `_min_max_scaling` from EIS Toolkit although I am not 100% sure. Please check and if it fits, we could replace this piece of code with a call to that function to have the operation centralized in one place only.
> 
> There's also the point that if we extend the scaling options in the future, it is good to use our own `transformations` module wherever we can

Hi niko,
as per your previous comment, I have used the '_min_max_scaling' instead of 'min_max_scaling.py.'

Reply:
Hi,
Ah yes you're right, I suggested to use `_min_max_scaling` initially – that would work just as well (also using range `(1, 0)`). I do not know why you are getting incorrect results, but if you run the scaling as the last operation, the data should definitely end up in the given range and not 0-25 like in the image you attached. I am able to produce the correct proximity computation result with this method that I suggest. You can push your new implementation to GitHub and I can take a look if you want.

Reply:
Hi,
I did find the problem, there was a naming error in my code, I used the same variable to get maximum distance and for scaling . Due to which the value for scaling was overridden. I found it. Also, just a small query I have here, for log scaling, there are various methods (log to base 10/ natural log), should there be another variable that the user must provide inorder to select the base of the log?

Reply:
Hi, ok good! Not another variable I would say, rather you can include the log variations in the scaling parameter like this `Literal['linear', 'ln', 'log2', 'log10']`